lib/cli.js

```diff
@@ -0,0 +1,201 @@
+        -p, --parameters        JSON string of parameters to override on create/update
+        -a, --all               Toboggan deploy of all functions found in subdirectories
+    `
+  }, {
+    alias: {
+      c: 'config-bucket',
+      t: 'template-bucket',
+      n: 'name',
+      r: 'region',
+      k: 'kms',
+      f: 'force',
+      e: 'extended',
+      d: 'decrypt',
+      p: 'parameters',
+      a: 'all'
+    },
+    string: ['config-bucket', 'template-bucket', 'name', 'region', 'parameters'],
+    boolean: ['force', 'extended', 'decrypt', 'all'],
+    default: {
+      region: 'us-east-1',
+      force: false,
+      kms: false,
+      extended: false,
+      decrypt: false,
+      parameters: undefined,
+      all: false
+    }
+  });
+
+  // if (cli.flags.true) {
+  //   if (create) {
+  //     // find all functions that don't have a matching stack
+  //   } else {
+  //     // find all stacks that do have a matching stack
+  //   }
+  // }
+
+
+  // check the caller location and set sane template default location
+  if (!cli.flags.all) { // no tobogganing
+    if (!cli.input[2] && cli.input[0] != 'init') {
+      var files = fs.readdirSync(process.cwd());
+      if (files.indexOf('function.template.js') === -1) {
+        throw new Error('function.template.js file not found in ' + process.cwd());
+      }
+      cli.input[2] = './function.template.js';
+      // generate default function name of repoDir-functionDir
+      // stack name then is later set to repoDir-functionDir-environment
+      if (!cli.flags.name) {
+        cli.flags.name = cli.flags.n = (process.cwd().split(path.sep).slice(-2)).join('-');
+      }
+    }
+  }
+
+  if (/init/.test(cli.input[0])) {
+    if (!cli.input[1])
+      throw new Error('Please provide a function name to initialize');
+    if (cli.flags.all)
+      throw new Error('init tobogganing not currently supported');
+  }
+
+  // Make sure that we can set a template bucket for create and update commands
+  if (/create|update/.test(cli.input[0])) {
+    if (!cli.flags.templateBucket && !env.AWS_ACCOUNT_ID)
+      throw new Error('Provide $AWS_ACCOUNT_ID as an environment variable to use the default template bucket, or set --template-bucket');
+    if (!cli.flags.templateBucket) cli.flags.templateBucket =
+      ['cfn-config-templates', env.AWS_ACCOUNT_ID, cli.flags.region].join('-');
+  }
+  // Make sure that we can set a configs bucket for create and update commands
+  if (/create|update|save/.test(cli.input[0])) {
+    if (!cli.flags.configBucket && !env.AWS_ACCOUNT_ID)
+      throw new Error('Provide $AWS_ACCOUNT_ID as an environment variable to use the default config bucket, or set --config-bucket');
+    if (!cli.flags.configBucket) cli.flags.configBucket =
+      ['cfn-configs', env.AWS_ACCOUNT_ID, cli.flags.region].join('-');
+  }
+
+  var parameters = {};
+
+  if (cli.flags.parameters) {
+    try {
+      parameters =
```

Comment:
Ah ok! The linebreaks confused me and made me think these were unset variables. No need to worry about these lines then.